hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/RowKeyGeneratorHelper.java

```diff
@@ -116,6 +119,7 @@ public static String getPartitionPathFromRow(Row row, List<String> partitionPath
         }
       } else { // nested
         Object nestedVal = getNestedFieldVal(row, partitionPathPositions.get(field));
+        nestedVal = convertInstantToTimestampIfNeed(nestedVal);
```

Comment:
try to be consistent with the other case

```suggestion
        Object data = getNestedFieldVal(row, partitionPathPositions.get(field));
        nestedVal = convertInstantToTimestampIfNeed(data);
```